cf-keycrypt/cf-keycrypt.c

```diff
@@ -0,0 +1,388 @@
+/*
+   cf-keycrypt.c
+
+   Copyright (C) 2017 cfengineers.net
+
+   Written and maintained by Jon Henrik Bjornstad <jonhenrik@cfengineers.net>
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+#include <config.h>
+#include <cf3.defs.h>
```

Comment:
Always first inclusion must be 
* either `platform.h`, which itself includes `config.h`.
* OR `cf3.defs.h` which includes itself `platform.h` and a bunch of libpromises stuff.

I think the 2nd is what you want here.